### PR TITLE
mon: allow cluter and debug logs to go to stderr, with appropriate prefix

### PR DESCRIPTION
**Pull request by liewegas**
This would let us send both the debug log  and the mon's cluster logs to stderr, differentiated
by a prefix string.

### CODE REVIEW

File: src/mon/LogMonitor.cc [hunk 1]
```diff
@@ -116,6 +116,10 @@ void LogMonitor::update_from_paxos(bool *need_bootstrap)
       if (channel.empty()) // keep retrocompatibility
         channel = CLOG_CHANNEL_CLUSTER;
 
+      if (g_conf->mon_cluster_log_to_stderr) {
```
Comment (tchaikov) [inline src/mon/LogMonitor.cc]: need to use `get_val<>()` .

File: src/log/Log.h [hunk 2]
```diff
@@ -70,6 +72,7 @@ class Log : private Thread
   void set_log_file(std::string fn);
   void reopen_log_file();
   void chown_log_file(uid_t uid, gid_t gid);
+  void set_log_stderr_prefix(const std::string& p);
```
Comment (tchaikov) [inline src/log/Log.h]: the function signature in header file does not match that in .cc file. see https://github.com/ceph/ceph/pull/19385/files#diff-8bc11b017f7a3710bf6737063a301577R136

**Comment by tchaikov** [comment]
i am not sure what is the usecase of sending error messages to stderror if not for debugging purpose, because the process closes stdout/stderr when it is daemonizing itself. but the change looks sane.

**Comment by liewegas** [comment]
It's an alternative approach to log collection for container environments, where the daemon is run with -f and stdout/stderr go to fluentd and into elasticsearch.  For the mons, we would prefix debug logs with "debug " and the mon logs are prefixed by teh channel name, allowing the log stream to be differentiated.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-12-15_11:03:39-rados-wip-kefu-testing-2017-12-15-1528-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-12-16_03:28:12-rados-wip-kefu-testing-2017-12-15-1528-distro-basic-mira/

**Pull request merged**